pkg/dss/scd/store.go

```diff
@@ -2,14 +2,34 @@ package scd
 
 import (
 	"context"
+	"time"
 
 	"github.com/golang/geo/s2"
 	dssmodels "github.com/interuss/dss/pkg/dss/models"
 	scdmodels "github.com/interuss/dss/pkg/dss/scd/models"
 )
 
-// Store abstracts interactions with a backing data store.
-type Store interface {
+// OperationStore abstracts operation-specific interactions with the backing data store.
+type OperationStore interface {
+	// GetOperation returns the operation identified by "id".
+	GetOperation(ctx context.Context, id scdmodels.ID) (*scdmodels.Operation, error)
+
+	// DeleteOperation deletes the operation identified by "id" and owned by "owner".
+	// Returns the deleted Operation and all Subscriptions affected by the delete.
+	DeleteOperation(ctx context.Context, id scdmodels.ID, owner dssmodels.Owner) (*scdmodels.Operation, []*scdmodels.Subscription, error)
```

Comment:
I know this is how the Subscription storage API methods are constructed, but I think it's probably a better idea to reduce the application logic in the storage layer, so the application layer should check the owner rather than the storage layer.  Concretely, I suggest removing the owner argument.

This also leads into the question of how we want to handle deletion requests for non-existent Operations.  I suggest the application layer should query the Operation first, then check whether it exists and who the owner is, then delete the Operation.  With this pattern, there is certainly no need for the owner argument.

Reply:
As pointed out before, we designed the storage layer interfaces to allow for richer storage layer implementations such that we avoid exposing the notion of transactions to service code.

Reply:
That would introduce a roundtrip to the storage layer for hardly any gain from my POV. In particular, we can easily check if the number of affected rows in the database is 0 and return an error in that case.